file_catalog/validation.py

```diff
@@ -5,56 +5,56 @@ def is_valid_sha512(hash_str):
     """Checks if `hash_str` is a valid SHA512 hash"""
     return re.match(r"[0-9a-f]{128}", str(hash_str), re.IGNORECASE) is not None
 
-def has_forbidden_attributes_creation(apihandler, metadata):
+def has_forbidden_attributes_creation(apihandler, config, metadata):
     """
     Checks if dict (`metadata`) has forbidden attributes.
 
     Returns `True` if it has forbidden attributes.
     """
 
-    if set(('_id', 'mongo_id')) & set(metadata):
+    if set(config.get_list('metadata', 'forbidden_fields_creation')) & set(metadata):
```

Comment:
Because of the fact that the config can't change while we're running, I might be inclined to package up the validation methods into a class, which takes the config as input.  That way it can store these lists locally instead of directly referencing the config each time.